source/SharedCrtResourceManager.cpp

```diff
@@ -211,7 +211,7 @@ int SharedCrtResourceManager::establishConnection(const PlainConfig &config)
     if (!connection->Connect(config.thingName->c_str(), true, 0))
     {
         LOGM_ERROR(TAG, "MQTT Connection failed with error: %s", ErrorDebugString(connection->LastError()));
-        return connection->LastError();
+        return RETRY;
```

Comment:
Why do we want to retry establishing connection over here?

Reply:
At this point we are not dealing with a client configuration error, we're dealing with a problem connecting to IoT Core. This could be caused by network adapters, network issues, API availability, or improperly configured IoT policies/certificates. Since these issues are all external to the Device Client, we attempt retry in the hope that the problem will be mitigated.